### PR TITLE
Refactor NonZeroLayerTest, NormalizeL2LayerTest, OneHotLayerTest, PadLayerTest

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Refactor NonZeroLayerTest, NormalizeL2LayerTest, OneHotLayerTest, PadLayerTest

### Tickets:
 - [CVS-111364](https://jira.devtools.intel.com/browse/CVS-111364)

### CODE REVIEW

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/one_hot.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        ov::element::Type,          // On & Off values type (any supported type)
+        float,                      // OnValue
+        float,                      // OffValue
+        int64_t,                    // axis
+        ov::element::Type,          // Model type
+        std::vector<InputShape>,    // Input shapes
+        std::string                 // Target device name
+> oneHotLayerTestParamsSet;
+
+class OneHotLayerTest : public testing::WithParamInterface<oneHotLayerTestParamsSet>,
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/one_hot.hpp:27]: minor, align the inheritance

File: src/tests/functional/shared_test_classes/src/single_op/pad.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    result << "TrgDev=" << target_device;
+    return result.str();
+}
+
+std::shared_ptr<ov::Node> PadLayerTest::create_pad_op(const ngraph::Output<ov::Node>& data,
+                                    const std::vector<int64_t>& pads_begin,
+                                    const std::vector<int64_t>& pads_end,
+                                    float arg_pad_value,
+                                    ov::test::utils::PadMode pad_mode) const {
+    return ngraph::builder::makePad(data, pads_begin, pads_end, arg_pad_value, pad_mode, false);
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/single_op/pad.cpp:53]: Why do we need this additional meta function?

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/one_hot.hpp (hunk 1 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/one_hot.hpp:27]: done

File: src/tests/functional/shared_test_classes/src/single_op/pad.cpp (hunk 2 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_op/pad.cpp]: got rid of builder func.
Not it only contain a create_pad_op method to create appropriate operation: pad_v1 for PadLayerTest and pad_v12 for Pad12LayerTest

**Review by iefode — APPROVED** [review]

**Pull request merged**